backend/app/extraction/ExternalTranscriptionWorker.scala

```diff
@@ -4,63 +4,91 @@ import cats.syntax.either._
+      new ReceiveMessageRequest(transcribeConfig.transcriptionOutputQueueUrl)
+        .withMaxNumberOfMessages(10)
+        .withAttributeNames("MessageGroupId", "ApproximateReceiveCount")
     ).getMessages
 
     if (messages.size() > 0)
       logger.info(s"retrieved ${messages.size()} messages from queue Transcription Output Queue")
-    else logger.info("No message found")
+    else
+      logger.info("No message found")
 
     messages.asScala.toList.foldLeft(0) { (completed, message) =>
-      val result = for {
-        transcriptionOutput <- parseMessage[TranscriptionOutput](message)
-        transcription <- blobStorage.get(transcriptionOutput.outputBucketKeys.text)
-        txt = new String(transcription.readAllBytes(), StandardCharsets.UTF_8)
-        _ <- addDocumentTranscription(transcriptionOutput, txt)
-        _ <- markAsComplete(transcriptionOutput.id, "ExternalTranscriptionExtractor")
-      } yield {
-        amazonSQSClient.deleteMessage(
-          new DeleteMessageRequest(transcribeConfig.transcriptionOutputQueueUrl, message.getReceiptHandle)
-        )
-        logger.debug(s"deleted message for ${transcriptionOutput.id}")
-      }
-
-      result match {
-        case Right(_) =>
-          completed + 1
-        case Left(failure) =>
-          logger.error(s"failed to process sqs message, ${failure.msg}", failure.toThrowable)
+      getMessageAttribute(message) match {
+        case Right(messageAttributes) =>
+          handleMessage(message, messageAttributes, completed)
+        case Left(error) =>
+          logger.error(s"Could not get message attributes from transcription output message, therefore can not update extractor. Message id: ${message.getMessageId}", error)
           completed
       }
     }
   }
 
-  private def markAsComplete(id: String, extractorName: String) = {
+
+  private def handleMessage(message: Message, messageAttributes: TranscriptionMessageAttribute, completed: Int) = {
+    val result = for {
+      transcriptionOutput <- parseMessage(message)
+      transcription <- blobStorage.get(transcriptionOutput.outputBucketKeys.text)
+      txt = new String(transcription.readAllBytes(), StandardCharsets.UTF_8)
+      _ <- addDocumentTranscription(transcriptionOutput, txt)
+      _ <- markExternalExtractorAsComplete(transcriptionOutput.id, "ExternalTranscriptionExtractor")
+    } yield {
+      amazonSQSClient.deleteMessage(
+        new DeleteMessageRequest(transcribeConfig.transcriptionOutputQueueUrl, message.getReceiptHandle)
+      )
+      logger.debug(s"deleted message for ${transcriptionOutput.id}")
+    }
+
+    result match {
+      case Right(_) =>
+        completed + 1
+      case Left(failure) =>
+        logger.error(s"failed to process sqs message", failure.toThrowable)
+        if (messageAttributes.receiveCount > 2) {
+          markAsFailure(new Uri(messageAttributes.messageGroupId), "ExternalTranscriptionExtractor", failure.msg)
+        }
```

Comment:
I think this handler needs updating to deal with the case where the message was processed succsesfully but there was a failure message

Reply:
I updated the failure cases to handle the error differently when the message is a failure one. This happens in the failure message scenario:
- move message to dead letter queue
- delete message from output queue
- mark blob/extractor relationship as failure